packages/server/src/mL/Trainer.ts

```diff
@@ -151,17 +151,35 @@ export default class Trainer {
 
     const metricsContainer = await this.docker.createContainer(this.project, Trainer.images.metrics, ["6006/tcp"]);
     this.container = await this.docker.createContainer(this.project, Trainer.images.train);
+
     await metricsContainer.start();
     await this.docker.runContainer(this.container);
+
     await metricsContainer.stop();
     await metricsContainer.remove();
+
+    this.status = TrainStatus.Stopped;
+  }
+
+  /**
+   * Calls updateCheckpoints every 10 seconds until training is stopped.
+   */
+  public async startCheckpointRoutine(): Promise<void> {
+    while (this.status !== TrainStatus.Stopped) {
+      await new Promise((resolve) => setTimeout(resolve, 10000));
+      await this.updateCheckpoints();
+    }
+    /* wait for any late metrics container parse and update one last time */
+    await new Promise((resolve) => setTimeout(resolve, 10000));
```

Comment:
```suggestion
    const UPDATE_INTERVAL = 10000;
    while (this.status !== TrainStatus.Stopped) {
      await new Promise((resolve) => setTimeout(resolve, UPDATE_INTERVAL));
      await this.updateCheckpoints();
    }
    /* wait for any late metrics container parse and update one last time */
    await new Promise((resolve) => setTimeout(resolve, UPDATE_INTERVAL));
```
Minimize magic numbers